services/shhext/service.go

```diff
@@ -37,28 +42,35 @@ type EnvelopeEventsHandler interface {
 
 // Service is a service that provides some additional Whisper API.
 type Service struct {
-	w            *whisper.Whisper
-	tracker      *tracker
-	nodeID       *ecdsa.PrivateKey
-	deduplicator *dedup.Deduplicator
-	debug        bool
+	w              *whisper.Whisper
+	tracker        *tracker
+	nodeID         *ecdsa.PrivateKey
+	deduplicator   *dedup.Deduplicator
+	protocol       *chat.ProtocolService
+	debug          bool
+	dataDir        string
+	installationID string
+	pfsEnabled     bool
 }
 
 // Make sure that Service implements node.Service interface.
 var _ node.Service = (*Service)(nil)
 
-// New returns a new Service.
-func New(w *whisper.Whisper, handler EnvelopeEventsHandler, db *leveldb.DB, debug bool) *Service {
+// New returns a new Service. dataDir is a folder path to a network-independent location
+func New(w *whisper.Whisper, handler EnvelopeEventsHandler, db *leveldb.DB, dataDir string, installationID string, debug bool, pfsEnabled bool) *Service {
```

Comment:
Maybe it would be better to pass at least some of these arguments in a struct? The code that calls this function is not very readable currently.